pym/bob/scm/url.py

```diff
@@ -788,6 +788,11 @@ def __getExtractor(self):
             raise ParseError("Invalid extract mode: " + self.__extract)
         return extractor
 
+    def postAttic(self, workspace):
+        if self.__separateDownload:
+            downloadDestination = os.path.abspath(os.path.join(workspace, "..", "download", self.__dir))
```

Comment:
Is the `os.path.abspath` really necessary? If true, then I would appreciate a comment why. It's somewhat unexpected.

Reply:
the reason for `abspath` was, that `os.path.exists` has some trouble if the path contains `..`:

```
>>> import os
>>> os.path.exists('foo/bar/../..')
False
``` 

however I think using `os.path.normpath` instead of `abspath` is the cleaner solution?

Reply:
I think the trouble comes from the fact that it fails if some of the path elements does not exist. Like for example if the user removed `workspace` from `dev/[...]/1/workspace/../download` then the path really does not exist. So I think you're right, `os.path.normpath` is more appropriate. Still, I would appreciate a comment because it's not obvious. :wink:

Reply:
This sounds like a good explanation :) I looked into the documentation but couldn't find anything for `os.path.exists`. But I found a [note](  https://docs.python.org/3/library/os.html#os.makedirs) for `os.makedirs` saying it's also confused by `..`. So I added `normalize` there as well. And a comment ;)

Reply:
:+1: